### PR TITLE
Add ts::hex and ts::unhex

### PR DESCRIPTION
**Pull request by cmcfarlen**

### CODE REVIEW

**Comment by zwoop** [comment]
Ship it.

**Comment by zwoop** [comment]
For this use case at least, this is at least as fast as Boost, sometimes even faster.

**Review by zwoop — APPROVED** [review]

File: include/tsutil/StringConvert.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#include <charconv>
+#include <stdexcept>
+#include <string>
+#include <system_error>
+
+namespace ts
+{
+
+inline std::string
+hex(const std::string_view input)
```
Comment (bneradt) [inline include/tsutil/StringConvert.h:30]: I was initially confused, assuming `hex` converted from base 10 or some such to hex. Maybe comment with something like:

```cpp
/** Convert values into their hex ascii equivalents.
 * @note This is a reimplementation of boost's hex.
 * @return The series of hex values corresponding to @a input
 */
```

File: include/tsutil/StringConvert.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+      throw std::runtime_error(std::make_error_code(err).message().c_str());
+    }
+  }
+
+  return result;
+}
+
+inline std::string
+unhex(const std::string_view input)
```
Comment (bneradt) [inline include/tsutil/StringConvert.h:49]: Maybe doc with:

```cpp
/** The inverse function of hex above.
 * @note This is a reimplementation of boost's unhex.
 */

**Review by bneradt — CHANGES_REQUESTED** [review]

**Comment by bneradt** [comment]
Looks good, other than the request for documentation. I verified in godbolt that hex this seems to act like boost's.

**Review by bneradt — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**